crates/js-component-bindgen/src/core.rs

```diff
@@ -347,7 +347,7 @@ impl Augmenter<'_> {
         // before.
         let mut types = TypeSection::new();
         for ty in self.types.iter() {
-            assert!(!ty.is_final);
+            // assert!(!ty.is_final);
```

Comment:
This assertion removal was necessary to get the upgrade tests to pass, but needs to be looked into further.

@alexcrichton would value your eyes on this if you have a moment.

Reply:
Ah here I think you can change `self.types.iter()` to [`into_iter_err_on_gc_types`](https://docs.rs/wasmparser/latest/wasmparser/type.TypeSectionReader.html#method.into_iter_err_on_gc_types) which should handle this for you now (that didn't exist when I first wrote this)

Reply:
Thanks.